src/aio/aio_provider.h

```diff
@@ -61,6 +61,8 @@ class aio_provider
 
     virtual error_code close(dsn_handle_t fh) = 0;
     virtual error_code flush(dsn_handle_t fh) = 0;
+    virtual error_code write(aio_context *aio_ctx, uint32_t *processed_bytes) = 0;
```

Comment:
`/*out*/ uint32_t *processed_bytes`

Reply:
Yes, we can do better by:
```suggestion
    virtual error_code write(const aio_context &aio_ctx, /*out*/ uint32_t *processed_bytes) = 0;
```